### PR TITLE
add lightgbm support

### PR DESCRIPTION
**Pull request by lukas**
Ported over basic lightgbm support.

### CODE REVIEW

File: wandb/__init__.py [hunk 1]
```diff
@@ -92,6 +92,7 @@ def _is_internal_process():
 sklearn = _lazyloader.LazyLoader('wandb.sklearn', globals(), 'wandb.sklearn')
 tensorflow = _lazyloader.LazyLoader('wandb.tensorflow', globals(), 'wandb.tensorflow')
 xgboost = _lazyloader.LazyLoader('wandb.xgboost', globals(), 'wandb.framework.xgboost')
+xgboost = _lazyloader.LazyLoader('wandb.lightgbm', globals(), 'wandb.framework.lightgbm')
```
Comment (raubitsj) [inline wandb/__init__.py:95]: typo

Comment (lukas) [reply wandb/__init__.py]: oops thanks

File: wandb/lightgbm/__init__.py~ [hunk 2]
```diff
@@ -0,0 +1,10 @@
+"""
```
Comment (raubitsj) [inline wandb/lightgbm/__init__.py~:1]: no ~ files please :)

Comment (lukas) [reply wandb/lightgbm/__init__.py~]: ugh sorry

**Pull request merged**